### PR TITLE
NEPT-1115: Implement the nexteuropa_varnish_prevent_purge parameter

### PR DESCRIPTION
**Pull request by gillesdeudon**

### CODE REVIEW

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.admin.inc [hunk 1]
```diff
@@ -12,17 +12,21 @@ function nexteuropa_varnish_admin_settings_form($form, $form_state) {
...
   $form['purge_cache'] = array(
     '#type' => 'fieldset',
     '#title' => t('Purge caches'),
     '#description' => t('Flush all caches (Drupal & Varnish)'),
   );
 
+  // If the purge mechanism is prevented to work, then the button must be
+  // disabled and a warning message must be displayed.
+  $disabled = _nexteuropa_varnish_temporary_message();
+
```
Comment (imanoleguskiza) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.admin.inc]: Minor thing: do we need to set this variable? We could just set the value on the array directly.

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module [hunk 2]
```diff
@@ -435,3 +443,46 @@ function _nexteuropa_varnish_get_node_paths($node) {
...
+ * Sets the temporary message when the purge mechanism is disabled by force.
+ *
+ * @return bool
+ *   TRUE if the message is set and then that means the mechanism is disabled by
+ *   force; otherwise FALSE.
+ */
+function _nexteuropa_varnish_temporary_message() {
+  if (_nexteuropa_varnish_prevent_purge()) {
+    drupal_set_message(
+      t('The purge mechanism is temporary disabled.<br /> Purge rules still can be created but they will not be executed until it is enabled again.'),
```
Comment (imanoleguskiza) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]: This should say "Purge rules can still be created ....."

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module [hunk 3]
```diff
@@ -435,3 +443,46 @@ function _nexteuropa_varnish_get_node_paths($node) {
 
   return $paths;
 }
+
+/**
+ * Determines if the purge mechanism disabling must be forced temporary.
```
Comment (imanoleguskiza) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]: I think the correct word is temporarily

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.admin.inc (hunk 1 above)
Comment (gillesdeudon) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.admin.inc]: I did like that for the clarity of the code.

**Comment by imanoleguskiza** [comment]
👍

File: profiles/common/modules/custom/nexteuropa_varnish/README.md [hunk 4]
```diff
@@ -136,3 +136,14 @@ certain paths.
 
 All of HTTP requests are send by the `_nexteuropa_varnish_purge_paths()`
 function.
+
+## Blocking temporary the purge mechanism
+
+It is always possible to block temporary the purge mechanism (via rules or via the "full" button).
+
+To do so, the following line must be added to the settings file:
+`$conf['nexteuropa_varnish_prevent_purge'] = TRUE;`
```
Comment (gillesdeudon) [inline profiles/common/modules/custom/nexteuropa_varnish/README.md]: No, I prefer my sentence as it mixes explanation and example

File: profiles/common/modules/custom/nexteuropa_varnish/README.md [hunk 5]
```diff
@@ -136,3 +136,14 @@ certain paths.
 
 All of HTTP requests are send by the `_nexteuropa_varnish_purge_paths()`
 function.
+
+## Blocking temporary the purge mechanism
+
+It is always possible to block temporary the purge mechanism (via rules or via the "full" button).
```
Comment (gillesdeudon) [inline profiles/common/modules/custom/nexteuropa_varnish/README.md]: Taken into account with small adaptations

**Comment by imanoleguskiza** [comment]
👍

**Comment by netlooker** [comment]
:+1:

**Pull request merged**